### DISCUSSION TITLE
With rabbitmq, is it possible to write routing_key of the message into the table?

### DISCUSSION THREAD

**Discussion by AntonOvsyannikov** [ClickHouse/ClickHouse#56349] [General]
Hello, considering we are using [this guide](https://clickhouse.com/docs/en/engines/table-engines/integrations/rabbitmq) to connect to rabbitmq and write messages from topic exchange with routing key like `foo.#` to our table. There is no routing_key in the message body, but routing_key itself is important part which should be stored. Are there any method to store the routing_key of the message into separate column?

**Comment by UnamedRus** [comment]
List of virtual columns available for RabbitMQ, if routing_key isn't here, you can make feature request to add it.

https://github.com/ClickHouse/ClickHouse/blob/0f76ba83e049d486e36c738abdf9d2ce1e2beb11/src/Storages/RabbitMQ/StorageRabbitMQ.cpp#L1220-L1227

**Reply by AntonOvsyannikov** [reply]
Thanks a lot! Will try!
